docs/guides/writing-relationships.md

```diff
@@ -0,0 +1,88 @@
+# Writing Relationships
+
+:::note
+Want to learn more about writing relationships to SpiceDB, the various strategies and their pros and cons? Read our [blog post about writing relationships]
+:::
+
+[blog post about writing relationships]: https://authzed.com/blog/writing-relationships-to-spicedb/
+
+In [SpiceDB], a permissions system is defined by two items: the [schema] which defines *how* data can be represented, and the *relationships*, defining the way the objects are actually related to one another.
+
+[schema]: /guides/schema
+[SpiceDB]: https://github.com/authzed/spicedb
+
+It is the application's responsibility to keep the relationships within SpiceDB up-to-date and reflecting the state of the application; how an application does so can vary based on the specifics of the application, so below we outline a few approaches.
+
+## Two writes + commit
+
+The most common and straightforward way to store relationships in SpiceDB is to use a 2 phase commit-like approach, making use of a transaction from the relational database along with a [WriteRelationships] call to SpiceDB.
+
+[WriteRelationships]: https://buf.build/authzed/api/docs/main/authzed.api.v1#WriteRelationships
+
+```python title='Example of a 2PC-like approach'
+# Open the database transaction
+try:
+    with db.transaction() as transaction:
+        # Update the relationship in the database.
+        document = Document(
+            id="somedoc",
+            owner=some_user,
+        )
+        transaction.add(document)
+
+        # Add the relationship(s) in SpiceDB.
+        resp = client.WriteRelationships(...)
+
+        # Store the ZedToken we've received.
+        document.zedtoken = result.written_at
+
+        # Transaction is committed on close
+except:
+    # Delete the relationship(s) just written
+    client.DeleteRelationships(...)
+    raise
+```
+
+## Streaming commits
+
+Another approach is to stream updates to both a relational database and SpiceDB via a third party streaming system such as [Kafka], using a pattern known as [Command Query Responsibility Segregation] (CQRS)
+
+[Kafka]: https://kafka.apache.org/
+[Command Query Responsibility Segregation]: https://www.confluent.io/blog/event-sourcing-cqrs-stream-processing-apache-kafka-whats-connection/
+
+In this design, any updates to the relationships in both databases are published as **events** to the streaming service, with each event being consumed by a system which performs the updates in both the database and in SpiceDB.
+
+## SpiceDB-only relationships
```

Comment:
I might put this first, so people are primed to consider "do I really even need to copy this data?" before they start learning about actually copying.